lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1387,26 +1483,31 @@ class MongoClientInterface {
                 const obj = doc.value;
                 const objMetadata = new ObjectMD(obj.value);
                 objMetadata.setOriginOp('s3:ObjectRemoved:Delete');
-                objMetadata.setDeleted(true);
+                // Not supported in 7.x
+                // objMetadata.setDeleted(true);
                 return next(null, objMetadata.getValue());
             }),
             // We update the full object to get the whole object metadata
             // in the oplog update event
-            (objMetadata, next) => collection.bulkWrite([
-                {
+            (objMetadata, next) => {
+                const operations = [{
                     updateOne: {
                         filter: updateDeleteFilter,
                         update: {
                             $set: { _id: key, value: objMetadata },
                         },
                         upsert: false,
                     },
-                }, {
+                }];
+
+                // Add the delete operation if the shouldDelete flag is true
+                operations.push({
                     deleteOne: {
                         filter: updateDeleteFilter,
                     },
-                },
-            ], { ordered: true }, () => next(null)),
+                });
+                collection.bulkWrite(operations, { ordered: true }, () => next(null));
+            },
```

Comment:
no change here, esp. the introduced comment does not seem valid?
→ keep as the code was before...

Reply:
Reverted in https://github.com/scality/Arsenal/pull/2134/commits/cc75b89115a7a8d4f26417fde5103f80421bedef#diff-9c207befbd8f4d79a3b6e3d7d143444e804ca1684236b8fd9fa0f9e03c53477aR1494